# Taller MCDA para e-commerce
Nombre: ______________ &nbsp; Fecha: ________

## Actividad 1 – Límites de conocimiento

La organización no tiene toda la información sobre las plataformas, trabaja con evidencia limitada (costos, latencia, disponibilidad, encuestas TAM y una revisión ética/legal). Por eso es importante identificar qué cosas no se saben con certeza y qué se podría hacer para reducir esa incertidumbre:

| No se sabe con certeza | Evidencia para reducirlo |
|---|---|
| Costos de mantenimiento a 3 años | Cotizaciones y contratos del proveedor |
| Cómo se comporta la plataforma con más carga (ej. Cyber Day) | Pruebas de carga, SLA del proveedor |
| Si la persuasión realmente aumenta la conversión | Pruebas A/B en un piloto |

Hay que diferenciar lo que es un dato medido de lo que es una suposición. Por ejemplo, "la latencia medida fue 250 ms" es un dato que se observó, mientras que "la plataforma seguirá siendo rápida durante 3 años" es una suposición, porque depende de cosas que todavía no pasan (más usuarios, cambios en la carga, etc.). Lo mismo ocurre con el TCO: el valor que tenemos es una estimación, no un costo ya pagado.

## Actividad 2 – Formular la decisión

"Dadas las alternativas **ShopFast, TrustCart, LeanCommerce y ProCommerce**, la organización debe seleccionar **una plataforma de e-commerce** considerando **costo, aceptación de usuarios, desempeño técnico, persuasión responsable y cumplimiento ético/legal**."

La decisión no es solo técnica: involucra a distintas personas con intereses diferentes.

- Quién decide: la gerencia (comercial y TI).
- Quién la usará: los clientes y el personal interno (ventas, marketing, TI).
- Quién puede ser afectado: clientes, organización, equipo TI.
- Quién pone restricciones: área legal, regulador (SERNAC, Ley 19.496) y políticas internas.

Esto sirve para el resto del taller, porque cada uno de estos actores valora cosas distintas y eso después se convierte en criterios o restricciones del modelo.

## Actividad 3 – Árbol axiológico

Siguiendo a Keeney, se parte por los valores antes de mirar las alternativas. El valor superior es **V0: adopción responsable y sostenible de una solución de e-commerce**, que se divide en cuatro dimensiones: V1 valor para el usuario, V2 valor económico, V3 responsabilidad ética/jurídica y V4 sostenibilidad tecnológica. En la tabla se relaciona cada stakeholder con lo que valora y con un objetivo que lo represente:

| Stakeholder | ¿Qué valora? | Objetivo |
|---|---|---|
| Cliente | Comprar fácil y que no lo engañen | Maximizar PU y PEOU (V1) |
| Organización | Rentabilidad y ventas | Minimizar TCO (V2) |
| Equipo tecnológico | Estabilidad y fácil mantenimiento | Maximizar disponibilidad y mantenibilidad (V4) |
| Área legal/compliance | Cumplir la ley | Cumplir normativa de consumidor y datos (V3) |
| Sociedad/regulador | Comercio justo y seguro | Respetar autonomía y seguridad (V3) |

Los objetivos de V1, V2 y V4 se pueden medir y comparar entre plataformas, por lo que pasan a ser criterios con peso. Los de V3 no deberían negociarse, por eso más adelante se usan como gates (restricciones).

## Actividad 4 – Dato y constructo

Escribir `tam = {"A": 9, "B": 8, "C": 7, "D": 9}` no basta porque son números sin respaldo: no se sabe quién los puso ni cómo se obtuvieron, por lo que se pueden calcular pero no son creíbles. Además, TAM tiene dos constructos distintos, la utilidad percibida (PU) y la facilidad de uso percibida (PEOU), y en ese diccionario quedan mezclados en un solo valor, perdiendo información (una plataforma puede ser útil pero difícil de usar).

Según Davis, PU y PEOU se miden preguntándole a los usuarios. Para considerar los valores creíbles se necesitaría:

- el cuestionario usado (ítems PU1–PU3 y PEOU1–PEOU3, escala Likert 1 a 5);
- cuántos usuarios respondieron y si eran representativos de los clientes reales;
- la tarea que hicieron antes de responder (por ejemplo, una compra de prueba);
- la fecha de la medición y quién la hizo.

## Actividad 5 – Técnica persuasiva: escasez ("Quedan 3 unidades")

Elegí la técnica de escasez, que consiste en avisarle al cliente que quedan pocas unidades de un producto. Para revisar si es responsable o manipuladora se usa el criterio del taller: persuasión responsable = influencia + veracidad + autonomía + transparencia.

| Pregunta | Respuesta |
|---|---|
| ¿Qué conducta facilita? | Que el cliente compre sin postergar |
| ¿Qué evidencia muestra? | El stock disponible |
| ¿Puede ser rechazada? | Sí, el usuario puede ignorarlo |
| ¿Conserva la autonomía? | Sí, si el stock es real |
| ¿Qué la haría manipulación? | Mostrar "quedan 3" cuando hay 742 (falsa escasez) |

En conclusión, la escasez es responsable solo si el número que se muestra es real y el cliente puede ignorarlo. Si el dato es falso, pasa a ser manipulación porque se influye en la compra usando un engaño. Este tipo de prácticas es lo que se revisa en el gate de autonomía.

## Actividad 6 – Flujo de datos "Productos para ti"

Según Nissenbaum, no basta con preguntar si se usan datos personales, sino que hay que ver quién los usa, qué datos, para qué, con quién se comparten y bajo qué norma.

1. Datos: historial de compras, productos vistos y búsquedas.
2. Los recolecta: la tienda a través de la plataforma.
3. Finalidad: recomendar productos de interés.
4. Recibe la inferencia: el mismo cliente.
5. ¿Es esperable? Sí, mientras los datos no se compartan con terceros ni se usen para otra cosa.
6. Control del usuario: poder desactivar la personalización y borrar su historial.

El flujo es adecuado mientras los datos se queden dentro de la tienda y se usen solo para recomendar, que es lo que un cliente espera. Si se vendieran a otras empresas o se usaran para otro fin, se rompería la integridad contextual. En cuanto a la norma, en Chile aplica la ley de protección de datos personales vigente; la Ley 21.719 entra en vigencia el 1 de diciembre de 2026, por lo que todavía es normativa futura.

## Actividad 7 – Gate de admisibilidad

Algunos valores no deberían poder compensarse con un buen puntaje en otros criterios. Por eso se definen cuatro condiciones obligatorias, y una alternativa solo es admisible si cumple todas:

ADM = Legal ∧ Autonomía ∧ Privacidad ∧ Seguridad

- **Legal:** cumple la ley del consumidor (Ley 19.496) y el Reglamento de Comercio Electrónico.
- **Autonomía:** no usa técnicas manipulativas como falsa escasez o temporizadores falsos.
- **Privacidad:** trata los datos personales según la normativa y con fines claros.
- **Seguridad:** protege los datos y los pagos de los clientes.

Con los datos del caso, la clasificación queda así:

| Alternativa | Legal | Autonomía | Privacidad | Seguridad |
|---|---|---|---|---|
| ShopFast | PASS | **FAIL** | PASS | PASS |
| TrustCart | PASS | PASS | PASS | PASS |
| LeanCommerce | PASS | PASS | PASS | PASS |
| ProCommerce | PASS | PASS | PASS | PASS |

ShopFast es la única que falla, en autonomía. En este caso ninguna queda INDETERMINATE porque los datos vienen como verdadero/falso, pero si faltara evidencia para algún gate (por ejemplo, no hay auditoría de seguridad) se marcaría INDETERMINATE y se trataría como FAIL hasta poder verificarlo.

## Paso 9 – Datos

In [1]:
import pulp

alternativas = {
    "A_ShopFast": {"tco": 35000, "latencia": 250, "disponibilidad": 99.70, "PU": 4.4, "PEOU": 4.0,
                   "mantenibilidad": 7.0, "persuasion_support": 10.0,
                   "legal": True, "autonomia": False, "privacidad": True, "seguridad": True},
    "B_TrustCart": {"tco": 45000, "latencia": 300, "disponibilidad": 99.90, "PU": 4.5, "PEOU": 4.6,
                    "mantenibilidad": 8.0, "persuasion_support": 7.5,
                    "legal": True, "autonomia": True, "privacidad": True, "seguridad": True},
    "C_LeanCommerce": {"tco": 30000, "latencia": 420, "disponibilidad": 99.50, "PU": 4.0, "PEOU": 4.2,
                       "mantenibilidad": 7.5, "persuasion_support": 6.5,
                       "legal": True, "autonomia": True, "privacidad": True, "seguridad": True},
    "D_ProCommerce": {"tco": 52000, "latencia": 210, "disponibilidad": 99.95, "PU": 4.7, "PEOU": 4.1,
                      "mantenibilidad": 8.5, "persuasion_support": 8.0,
                      "legal": True, "autonomia": True, "privacidad": True, "seguridad": True},
}

## Paso 10 – Funciones de valor

In [2]:
def valor_menor_mejor(x, minimo, maximo):
    valor = 10 * (maximo - x) / (maximo - minimo)
    return max(0, min(10, valor))

def valor_mayor_mejor(x, minimo, maximo):
    valor = 10 * (x - minimo) / (maximo - minimo)
    return max(0, min(10, valor))

def calcular_valores(alternativas):
    valores = {}
    for nombre, d in alternativas.items():
        valores[nombre] = {
            "costo": valor_menor_mejor(d["tco"], 25000, 60000),
            "rendimiento": valor_menor_mejor(d["latencia"], 150, 600),
            "disponibilidad": valor_mayor_mejor(d["disponibilidad"], 99.0, 100.0),
            "PU": valor_mayor_mejor(d["PU"], 1, 5),
            "PEOU": valor_mayor_mejor(d["PEOU"], 1, 5),
            "mantenibilidad": d["mantenibilidad"],
            "persuasion_support": d["persuasion_support"]
        }
    return valores

valores = calcular_valores(alternativas)

## Paso 11 – Pesos

- Costo 0.15: importa, pero no debería ganar solo lo más barato.
- Rendimiento 0.10 y disponibilidad 0.10: todas las opciones están en rangos aceptables.
- PU 0.20 y PEOU 0.20: son lo más importante, si el cliente no acepta la plataforma no compra.
- Mantenibilidad 0.10: importa al equipo TI, pero el cliente no lo ve.
- Persuasión 0.15: se quiere conversión, pero solo responsable.

Los pesos los asigné como analista según lo que valora la organización en el caso.

In [3]:
pesos = {
    "costo": 0.15,
    "rendimiento": 0.10,
    "disponibilidad": 0.10,
    "PU": 0.20,
    "PEOU": 0.20,
    "mantenibilidad": 0.10,
    "persuasion_support": 0.15
}

assert abs(sum(pesos.values()) - 1.0) < 1e-9

## Paso 12 – Scoring compensatorio

In [4]:
def calcular_scores(valores, pesos):
    score = {}
    for nombre in valores:
        score[nombre] = sum(pesos[c] * valores[nombre][c] for c in pesos)
    return score

score = calcular_scores(valores, pesos)

print("SCORING SIN GATES")
for nombre, s in score.items():
    print(nombre, round(s, 3))

SCORING SIN GATES
A_ShopFast 7.949
B_TrustCart 7.685
C_LeanCommerce 7.011
D_ProCommerce 7.61


### Actividad 8
1. El score más alto es de ShopFast (7.949).
2. No respeta todos los valores: falla en autonomía.
3. No sería correcto, porque la autonomía del cliente no debería poder "comprarse" con menor costo o más conversión. Por eso se trata como veto.

## Paso 13 – Gate no compensatorio

In [5]:
def es_admisible(d):
    return d["legal"] and d["autonomia"] and d["privacidad"] and d["seguridad"]

admisibles = {nombre: es_admisible(d) for nombre, d in alternativas.items()}

print("ADMISIBILIDAD")
for nombre, ok in admisibles.items():
    print(nombre, "PASS" if ok else "FAIL")

ADMISIBILIDAD
A_ShopFast FAIL
B_TrustCart PASS
C_LeanCommerce PASS
D_ProCommerce PASS


## Pasos 14 a 16 – Modelo PuLP y resolución

In [6]:
def resolver(alternativas, score, admisibles):
    modelo = pulp.LpProblem("Seleccion_Plataforma_Ecommerce", pulp.LpMaximize)
    x = pulp.LpVariable.dicts("seleccionar", alternativas.keys(), cat="Binary")

    modelo += pulp.lpSum(score[a] * x[a] for a in alternativas)   # funcion objetivo
    modelo += pulp.lpSum(x[a] for a in alternativas) == 1          # elegir una
    for a in alternativas:                                         # vetos
        if not admisibles[a]:
            modelo += x[a] == 0

    modelo.solve(pulp.PULP_CBC_CMD(msg=False))
    print("ESTADO:", pulp.LpStatus[modelo.status])
    for a in alternativas:
        if x[a].value() == 1:
            print("SELECCIONADA:", a, "Score:", round(score[a], 3))
    return x

x = resolver(alternativas, score, admisibles)

ESTADO: Optimal
SELECCIONADA: B_TrustCart Score: 7.685


### Actividad 9
`Optimal` solo indica que PuLP encontró la mejor solución del **modelo** planteado. Ese resultado depende de la **evidencia** usada, de las **preferencias** (pesos) y de las **restricciones** que pusimos, así que no demuestra que sea la mejor opción en el mundo real.

## Paso 17 – Trazabilidad

In [7]:
print("DETALLE")
for a in alternativas:
    print("\n", a, "Admisible:", admisibles[a], "Score:", round(score[a], 3))
    for criterio in pesos:
        contribucion = pesos[criterio] * valores[a][criterio]
        print(criterio, "valor=", round(valores[a][criterio], 3), "contribucion=", round(contribucion, 3))

DETALLE

 A_ShopFast Admisible: False Score: 7.949
costo valor= 7.143 contribucion= 1.071
rendimiento valor= 7.778 contribucion= 0.778
disponibilidad valor= 7.0 contribucion= 0.7
PU valor= 8.5 contribucion= 1.7
PEOU valor= 7.5 contribucion= 1.5
mantenibilidad valor= 7.0 contribucion= 0.7
persuasion_support valor= 10.0 contribucion= 1.5

 B_TrustCart Admisible: True Score: 7.685
costo valor= 4.286 contribucion= 0.643
rendimiento valor= 6.667 contribucion= 0.667
disponibilidad valor= 9.0 contribucion= 0.9
PU valor= 8.75 contribucion= 1.75
PEOU valor= 9.0 contribucion= 1.8
mantenibilidad valor= 8.0 contribucion= 0.8
persuasion_support valor= 7.5 contribucion= 1.125

 C_LeanCommerce Admisible: True Score: 7.011
costo valor= 8.571 contribucion= 1.286
rendimiento valor= 4.0 contribucion= 0.4
disponibilidad valor= 5.0 contribucion= 0.5
PU valor= 7.5 contribucion= 1.5
PEOU valor= 8.0 contribucion= 1.6
mantenibilidad valor= 7.5 contribucion= 0.75
persuasion_support valor= 6.5 contribucion= 0.97

## Paso 18 – Verificación

In [8]:
print("Pesos suman 1:", abs(sum(pesos.values()) - 1) < 1e-9)
print("Una sola alternativa elegida:", sum(x[a].value() for a in alternativas) == 1)
print("Inadmisibles en 0:", all(x[a].value() == 0 for a in alternativas if not admisibles[a]))

Pesos suman 1: True
Una sola alternativa elegida: True
Inadmisibles en 0: True


### Actividad 10 – Verificación manual (TrustCart)

S(B) = 0.15·4.286 + 0.10·6.667 + 0.10·9.0 + 0.20·8.75 + 0.20·9.0 + 0.10·8.0 + 0.15·7.5 = **7.685**, igual al valor de Python.

**Validación:** el modelo es una simplificación. Faltan criterios como integración con medios de pago o escalabilidad, no sabemos con cuántos usuarios se midió TAM, y las anclas y escalas lineales son supuestos que habría que justificar mejor.

## Paso 19 – Sensibilidad del peso de PEOU

In [9]:
def ajustar_peso(base, criterio_objetivo, nuevo_peso):
    suma_restantes = sum(v for k, v in base.items() if k != criterio_objetivo)
    nuevos = {}
    for k, v in base.items():
        if k == criterio_objetivo:
            nuevos[k] = nuevo_peso
        else:
            nuevos[k] = (v / suma_restantes) * (1 - nuevo_peso)
    return nuevos

def sensibilidad(alternativas, valores, admisibles):
    for nuevo_peou in [0.10, 0.15, 0.20, 0.25, 0.30]:
        w = ajustar_peso(pesos, "PEOU", nuevo_peou)
        scores_escenario = calcular_scores(valores, w)
        candidatos = [a for a in alternativas if admisibles[a]]
        ganador = max(candidatos, key=lambda a: scores_escenario[a])
        print("PEOU=", nuevo_peou, "ganador=", ganador, "score=", round(scores_escenario[ganador], 3))

sensibilidad(alternativas, valores, admisibles)

PEOU= 0.1 ganador= D_ProCommerce score= 7.592
PEOU= 0.15 ganador= B_TrustCart score= 7.602
PEOU= 0.2 ganador= B_TrustCart score= 7.685
PEOU= 0.25 ganador= B_TrustCart score= 7.767
PEOU= 0.3 ganador= B_TrustCart score= 7.849


### Actividad 11 – Robustez

| Peso PEOU | Ganadora | ¿Cambió? |
|---|---|---|
| 0.10 | ProCommerce | Sí |
| 0.15 | TrustCart | No |
| 0.20 | TrustCart | No |
| 0.25 | TrustCart | No |
| 0.30 | TrustCart | No |

La recomendación es **sensible**. Cuando el peso de PEOU sube, TrustCart sigue ganando, pero si baja a 0.10 gana ProCommerce, y con 0.15 TrustCart gana por muy poco (7.602 contra 7.600 aprox.). Esto pasa porque TrustCart le gana a ProCommerce principalmente en facilidad de uso y costo, mientras que ProCommerce es mejor en casi todo lo demás. Entonces la elección depende bastante de cuánta importancia se le dé a la facilidad de uso.

## Actividad 12 – Conclusión

"Se recomienda **TrustCart** porque supera **los cuatro gates (legal, autonomía, privacidad y seguridad)**. Entre las alternativas admisibles obtiene **el mayor score (7.685)** bajo los pesos y funciones de valor definidos. La evidencia de aceptación proviene de **la medición TAM (PU 4.5, PEOU 4.6)**. La recomendación es **sensible** frente a variaciones de **el peso de PEOU**. Por ello, la conclusión se considera **justificable pero no robusta** bajo las condiciones evaluadas."

## Desafío final – Quinta plataforma (E_NovaShop)

In [10]:
alternativas["E_NovaShop"] = {"tco": 41000, "latencia": 270, "disponibilidad": 99.85, "PU": 4.3, "PEOU": 4.7,
                              "mantenibilidad": 7.8, "persuasion_support": 8.2,
                              "legal": True, "autonomia": True, "privacidad": True, "seguridad": True}

valores = calcular_valores(alternativas)
score = calcular_scores(valores, pesos)
admisibles = {nombre: es_admisible(d) for nombre, d in alternativas.items()}

print("SCORES")
for nombre, s in score.items():
    print(nombre, round(s, 3), "PASS" if admisibles[nombre] else "FAIL")
print()
x = resolver(alternativas, score, admisibles)
print()
sensibilidad(alternativas, valores, admisibles)

SCORES
A_ShopFast 7.949 FAIL
B_TrustCart 7.685 PASS
C_LeanCommerce 7.011 PASS
D_ProCommerce 7.61 PASS
E_NovaShop 7.908 PASS

ESTADO: Optimal
SELECCIONADA: E_NovaShop Score: 7.908

PEOU= 0.1 ganador= E_NovaShop score= 7.74
PEOU= 0.15 ganador= E_NovaShop score= 7.824
PEOU= 0.2 ganador= E_NovaShop score= 7.908
PEOU= 0.25 ganador= E_NovaShop score= 7.992
PEOU= 0.3 ganador= E_NovaShop score= 8.075


"Se recomienda **NovaShop** porque supera **los cuatro gates**. Entre las alternativas admisibles obtiene **el mayor score (7.908)** bajo los pesos y funciones de valor definidos. La evidencia de aceptación proviene de **la medición TAM (PU 4.3, PEOU 4.7)**. La recomendación es **robusta** frente a variaciones de **el peso de PEOU entre 0.10 y 0.30**. Por ello, la conclusión se considera **creíble** bajo las condiciones evaluadas."

## Preguntas de cierre

**1. ¿Por qué MCDA no comienza necesariamente por los pesos?**
Porque primero hay que saber qué se valora y qué criterios lo representan; los pesos vienen después.

**2. ¿Qué aporta Keeney a la construcción del modelo?**
Propone partir por los valores y objetivos, y ver las alternativas como medios para lograrlos.

**3. ¿Qué diferencia existe entre la perspectiva de Roy y una búsqueda de "óptimo absoluto"?**
Roy ve el modelo como apoyo a la decisión, construido con los participantes, no como algo que encuentra un óptimo objetivo.

**4. ¿Por qué PU y PEOU requieren evidencia empírica?**
Porque son percepciones de los usuarios y hay que medirlas, no inventarlas.

**5. ¿Qué aporta Fogg y qué limitación ética debe añadirse a su uso?**
Fogg explica cómo la tecnología influye en el comportamiento; la limitación es que esa influencia debe ser veraz y respetar la autonomía.

**6. ¿Cómo ayuda Value Sensitive Design a un e-commerce?**
Ayuda a considerar valores como privacidad y autonomía desde el diseño de cada funcionalidad.

**7. ¿Por qué privacidad puede tratarse como límite axiológico?**
Porque una falla de privacidad no se compensa con menor costo y además es una obligación legal.

**8. ¿Qué diferencia existe entre un criterio compensatorio y un veto?**
Un criterio compensatorio se puede compensar con otros; un veto descarta la alternativa sin importar su score.

**9. ¿Qué demuestra realmente PuLP cuando entrega Optimal?**
Que es la mejor solución del modelo formulado, no necesariamente de la realidad.

**10. ¿Qué diferencia existe entre verificar y validar?**
Verificar es revisar que el modelo esté bien implementado; validar es revisar si representa bien el problema real.

**11. ¿Por qué un análisis de sensibilidad aumenta la credibilidad de la conclusión?**
Porque muestra si la conclusión se mantiene cuando cambian los supuestos, como los pesos.

**12. ¿Cómo se relacionan Claim, Argument y Evidence?**
El claim es lo que se afirma, la evidencia lo respalda y el argumento explica por qué esa evidencia sostiene el claim.